mmdata.c

```diff
@@ -1098,6 +1131,34 @@ nmbrString *nmbrSpace(long n)
   return sout;
 }
 
+/* Search for the nth occurrence of string2 in string1 */
+long nmbrInstrN(long start_position, long occ, nmbrString *string1,
+  nmbrString *string2, long start2, long length2)
+{
+  if (start_position < 1) start_position = 1;
+  start_position--;
+  for(; occ > 0;occ--) {

```

Comment:
Optimization: Use a dedicated loop variable in for loops. A compiler will allocate a CPU register for that.
for (long o = occ + 1; --o > 0;) {... (replace occ with o in loop)...}
is how I would write it.

Info: There is a tiny semantic change in my example: If the memory model of the computer supports signed magnitude instead of two's complement AND occ is MAX_LONG then (occ + 1)-1 == occ may not hold. We can safely ignore this nowadays.

Reply:
`long` is a signed type, and signed overflow is UB, so I think that this change is legal by the spec.

Reply:
@digama0 Exactly.  Because occ+1 is UB when occ==MAX_LONG, --o may contain anything, and the loop starts with a random value.  This cannot happen with the original code.  We can safely ignore this semantic difference here.